Automattic/wp-calypso
# chore: move storybook config to a separate package

### Changes proposed in this Pull Request

Moves Storybook config to a separate package, `@automattic/calypso-storybook`.

It includes the following improvements:

- Packages with a Storybook use a shared Babel cache, improving transpliation of common parts (eg: storybook itself).
- Deprecated the root scripts `yarn <package>:storybook:start` in favor of `yarn workspace <package> run storybook`.
- It is also possible to do `cd <package> && yarn storybook`. The package is more contained.
- No need to have storybook specific config in `@automattic/calypso-babel`. Now the Storybook config will inherit from `@automattic/calypso-babel` and make the necessary tweaks.


### Testing instructions

Run the storybook commands and verify Storybook loads and there are no warnings in the system console.
